src/node/plugins/staticDataPlugin.ts

```diff
@@ -122,44 +123,27 @@ export const staticDataPlugin: Plugin = {
     }
   },
 
-  transform(_code, id) {
-    if (server && loaderMatch.test(id) && '_importGlobMap' in server) {
-      // register this module as a glob importer
-      const { watch } = idToLoaderModulesMap[id]!
-      if (watch) {
-        ;(server as any)._importGlobMap.set(
-          id,
-          [Array.isArray(watch) ? watch : [watch]].map((globs) => {
-            const affirmed: string[] = []
-            const negated: string[] = []
-
-            for (const glob of globs) {
-              ;(glob[0] === '!' ? negated : affirmed).push(glob)
-            }
-            return { affirmed, negated }
-          })
-        )
-      }
-    }
-    return null
-  },
+  hotUpdate(ctx) {
+    if (ctx.type === 'update') return
 
-  handleHotUpdate(ctx) {
     const file = ctx.file
 
+    const modules: EnvironmentModuleNode[] = []
     // dependency of data loader changed
     // (note the dep array includes the loader file itself)
     if (file in depToLoaderModuleIdMap) {
       const id = depToLoaderModuleIdMap[file]!
       delete idToLoaderModulesMap[id]
-      ctx.modules.push(server.moduleGraph.getModuleById(id)!)
+      modules.push(this.environment.moduleGraph.getModuleById(id)!)
     }
 
     for (const id in idToLoaderModulesMap) {
       const { watch } = idToLoaderModulesMap[id]!
       if (watch && isMatch(file, watch)) {
-        ctx.modules.push(server.moduleGraph.getModuleById(id)!)
+        modules.push(this.environment.moduleGraph.getModuleById(id)!)
       }
     }
```

Comment:
The whole `_importGlobMap` part existed in `transform` hook is handled here. The `affirmed` / `negated` part is replaced with a simple `isMatch` because that's more aligned with [the `glob` call](https://github.com/vuejs/vitepress/blob/2e54970f7195c67b63908964575f589ce24b6d29/src/node/plugins/staticDataPlugin.ts#L103).